mapie/tests/test_time_series_regression.py

```diff
@@ -0,0 +1,357 @@
+    width_mean = (y_pis[:, 1, 0] - y_pis[:, 0, 0]).mean()
+    coverage = regression_coverage_score(
+        y_test, y_pis[:, 0, 0], y_pis[:, 1, 0]
+    )
+    np.testing.assert_allclose(width_mean, WIDTHS["prefit"], rtol=1e-2)
+    np.testing.assert_allclose(coverage, COVERAGES["prefit"], rtol=1e-2)
+
+
+def test_not_enough_resamplings() -> None:
+    """Test that a warning is raised if at least one residual is nan."""
+    with pytest.warns(UserWarning, match=r"WARNING: at least one point of*"):
+        mapie_ts_reg = MapieTimeSeriesRegressor(
+            cv=BlockBootstrap(n_resamplings=1, n_blocks=1), agg_function="mean"
+        )
+        mapie_ts_reg.fit(X, y)
+
+
+def test_no_agg_fx_specified_with_subsample() -> None:
+    """Test that a warning is raised if at least one residual is nan."""
+    with pytest.raises(
+        ValueError, match=r"You need to specify an aggregation*"
+    ):
+        mapie_ts_reg = MapieTimeSeriesRegressor(
+            cv=BlockBootstrap(n_resamplings=1, n_blocks=1),
+            agg_function=None,
+        )
+        mapie_ts_reg.fit(X, y)
+
+
+def test_invalid_aggregate_all() -> None:
+    """
+    Test that wrong aggregation in MAPIE raise errors.
+    """
+    with pytest.raises(
+        ValueError,
+        match=r".*Aggregation function called but not defined.*",
+    ):
+        aggregate_all(None, X)
+
+
+def test_aggregate_with_mask_with_prefit() -> None:
+    """
+    Test ``aggregate_with_mask`` in case ``cv`` is ``"prefit"``.
+    """
+    mapie_ts_reg = MapieTimeSeriesRegressor(cv="prefit")
+    with pytest.raises(
+        ValueError,
+        match=r".*There should not be aggregation of predictions if cv is*",
+    ):
+        mapie_ts_reg.aggregate_with_mask(k, k)
+
+    mapie_ts_reg = MapieTimeSeriesRegressor(agg_function="nonsense")
+    with pytest.raises(
+        ValueError,
+        match=r".*The value of self.agg_function is not correct*",
+    ):
+        mapie_ts_reg.aggregate_with_mask(k, k)
+
+
+def test_pred_loof_isnan() -> None:
+    """Test that if validation set is empty then prediction is empty."""
+    mapie_ts_reg = MapieTimeSeriesRegressor()
+    _, y_pred, _ = mapie_ts_reg._fit_and_predict_oof_model(
+        estimator=mapie_ts_reg,
+        X=X_toy,
+        y=y_toy,
+        train_index=[0, 1, 2, 3, 4],
+        val_index=[],
+    )
+    assert len(y_pred) == 0
+
+
+def test_MapieTimeSeriesRegressor_alpha_is_None() -> None:
+    """Test ``predict`` when ``alpha`` is None."""
+    mapie_ts_reg = MapieTimeSeriesRegressor(cv=-1).fit(X_toy, y_toy)
+
+    with pytest.raises(ValueError, match=r".*too many values to unpackt*"):
+        y_pred, y_pis = mapie_ts_reg.predict(X_toy, alpha=None)
+
+
+def test_MapieTimeSeriesRegressor_partial_fit_ensemble() -> None:
+    """Test ``partial_fit``."""
+    mapie_ts_reg = MapieTimeSeriesRegressor(cv=-1).fit(X_toy, y_toy)
+    assert round(mapie_ts_reg.conformity_scores_[-1], 2) == round(
+        np.abs(14.189 - 14.038), 2
```

Comment:
Should the results be included in a variable at the beginning of the file ?

Reply:
at the beginning of the file it would be hard to understand where they come from, I think

Reply:
This is what we do with the `COVERAGES` and `WIDTHS` variables.

Reply:
I added UPDATE_DATA and CONFORMITY_SCORES